### PR TITLE
[master] ZIL-4967: Fix Zil contract address calculation

### PR DESCRIPTION
**Pull request by JamesHinshelwood**
We also need to decrement the nonce for Zil-style transactions before calculating contract addresses.

### CODE REVIEW

**Review by n-hutton — APPROVED** [review]

**Comment by valeryz** [comment]
Is this a backport of 8.3 change?

**Comment by valeryz** [comment]
or, rather, not backport, but forward-port?

File: src/libServer/LookupServer.cpp [hunk 1]
```diff
@@ -558,6 +558,12 @@ Json::Value LookupServer::CreateTransaction(
 
     Transaction tx = JSONConversion::convertJsontoTx(_json);
 
+    if (tx.IsEth()) {
+      throw JsonRpcException(RPC_INVALID_PARAMETER,
+                             "Eth txs not supported for CreateTransaction api "
+                             "- use eth_sendRawTransaction");
+    }
+
```
Comment (valeryz) [inline src/libServer/LookupServer.cpp:566]: Yes! It is time! 

We had used this interface for EVM as well initially until we had eth_sendRawTransaction.

**Comment by yaron-zilliqa** [comment]
No need to raise 2 at the same time. You can raise one for the release branch and this will be carried forward by a merge from release -> master.

**Pull request merged**